vector/table_vector.go

```diff
@@ -0,0 +1,34 @@
+package vector
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// AllocateTableVector returns an allocation function that sorts the file on
+// the pos element with the sep separator.
+func AllocateTableVector(sep string, pos int) func(int) *Vector {
+	return func(size int) *Vector {
+		return &Vector{
+			s: make([]Element, 0, size),
+			NewElement: func(value string) Element {
+				num := strings.Split(value, sep)[pos]
```

Comment:
What about a regular expression to capture everything from the start of the line to the first \t. It will be constant time O(1). Because I think strings.Split is O(n)

Reply:
That's a great point. I will make changes.

Reply:
I came up with this regexp, but it made the operation slower.

```go
	str := fmt.Sprintf(`^([^%s]+%s){%d}([^%s]+)%s?`, sep, sep, pos, sep, sep)
	re := regexp.MustCompile(str)

	return func(size int) *Vector {
		return &Vector{
			s: make([]Element, 0, size),
			NewElement: func(value string) Element {
				num := re.FindStringSubmatch(value)[2]
				i, err := strconv.Atoi(num)
				if err != nil {
					panic(errors.Wrap(err, "converting value from string"))
				}
            }
		}
	}
}
```